include/fs/fs.h

```diff
@@ -170,20 +182,32 @@ struct fs_file_system_t {
 #define FS_SEEK_END	2	/* Seek from end of file.  */
 #endif
 
-
 /**
- * @brief File open
+ * @brief File open extended
  *
- * Opens an existing file or create a new one and associates
- * a stream with it.
+ * Opens or crates, if does not exist file, depending on mode provided
+ * and associates a stream with it.
  *
  * @param zfp Pointer to file object
  * @param file_name The name of file to open
+ * @param flags The mode flags, if not provided default to FO_CREATE | FO_RDWR
+ *
+ * @p flags can be empty, or combination of one or more of following flags:
+ *   FO_READ		open for read
+ *   FO_WRITE		open for write
+ *   FO_RDWR		open for read/write (<tt>FO_READ | FO_WRITE</tt>)
+ *   FO_CREATE		crete file if it does not exist
+ *   FO_APPEND		open file at the end
  *
  * @retval 0 Success
- * @retval -ERRNO errno code if error
+ * @retval -EINVAL when bad combination of flags is given
+ *	   -EINVAL when bad file name is given
+ *	   -NOENT  when file path is not possible (bad mount point)
+ *	   -ERRNO  other errno code, deending of file system backend
```

Comment:
> -ERRNO  other errno code, deending of file system backend

That's ambiguous/confusing (and with a few typos). Suggestion: "other negative error code, depending on file system backend".

Reply:
Also the way Doxygen works I believe `@retval` is required before each return value.  There's no implicit continuation onto subsequent lines.

Reply:
Will fix.

Reply:
Done.